hydromt_wflow/wflow.py

```diff
@@ -3147,37 +3147,48 @@ def setup_cold_states(
             self.set_config(option, states_config[option])
 
     # I/O
-    def read(self):
+    def read(
+        self,
+        config_fn: Union[Path, str] = None,
+        geom_fn: Union[Path, str] = "staticgeoms",
+    ):
         """Read the complete model schematization and configuration from file."""
-        self.read_config()
+        self.read_config(config_fn=config_fn)
         self.read_grid()
+        self.read_geoms(geom_fn=geom_fn)
+        self.read_forcing()
         self.read_intbl()
         self.read_tables()
-        self.read_geoms()
         self.read_states()
-        self.read_forcing()
         self.logger.info("Model read")
 
-    def write(self):
+    def write(
+        self,
+        config_fn: str = None,
+        grid_fn: Union[Path, str] = "staticmaps.nc",
+        geom_fn: Union[Path, str] = "staticgeoms",
+        forcing_fn: Union[Path, str] = None,
+    ):
```

Comment:
I think if you want to change any default argument of write, you should then call the single write methods one by one. The general write is only meant to be a shortcut for a quick write.

Reply:
I know, but I noticed while using it myself that sometimes I want to write everything but change one name of the output. I made it kwargs only to support the quick `model.write()`. I'd say, it's just a little bit of extra functionality.